src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -65,6 +69,102 @@ static char *IsolationLevelName[] = {
 };
 
 
+PG_FUNCTION_INFO_V1(citus_internal_start_management_transaction);
+PG_FUNCTION_INFO_V1(execute_command_on_other_nodes);
+PG_FUNCTION_INFO_V1(citus_mark_object_distributed);
+PG_FUNCTION_INFO_V1(commit_management_command_2PC);
+
+bool IsManagementCommand = false;
+FullTransactionId outerXid;
+static MultiConnection *managementCon = NULL;
+bool CitusManagementDatabase = true;
+
+Datum
+citus_internal_start_management_transaction(PG_FUNCTION_ARGS)
+{
+	outerXid = PG_GETARG_FULLTRANSACTIONID(0);
+	IsManagementCommand = true;
+
+	PG_RETURN_VOID();
+}
+
+
+Datum
+execute_command_on_other_nodes(PG_FUNCTION_ARGS)
+{
+	text *queryText = PG_GETARG_TEXT_P(0);
+	char *query = text_to_cstring(queryText);
+
+	List *queryList = NIL;
+	queryList = lappend(queryList, "SET citus.enable_create_role_propagation TO OFF;");
+	queryList = lappend(queryList, query);
+	queryList = lappend(queryList, "SET citus.enable_create_role_propagation TO ON;");
+	List *taskList = NodeDDLTaskList(NON_COORDINATOR_NODES, queryList);
+	DDLJob *ddlJob = NULL;
+	foreach_ptr(ddlJob, taskList)
+	{
+		ExecuteDistributedDDLJob(ddlJob);
+	}
+	PG_RETURN_VOID();
+}
+
+
+Datum
+citus_mark_object_distributed(PG_FUNCTION_ARGS)
+{
+	Oid classId = PG_GETARG_OID(0);
+	text *objectNameText = PG_GETARG_TEXT_P(1);
+	char *objectName = text_to_cstring(objectNameText);
+	Oid objectId = PG_GETARG_OID(2);
+	ObjectAddress *objectAddress = palloc0(sizeof(ObjectAddress));
+	ObjectAddressSet(*objectAddress, classId, objectId);
+	MarkObjectDistributedWithName(objectAddress, objectName);
+	PG_RETURN_VOID();
+}
+
+
+Datum
+commit_management_command_2PC(PG_FUNCTION_ARGS)
+{
+	CoordinatedRemoteTransactionsCommit();
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * Need to turn this to a correct function
+ */
+bool
+isCitusManagementDatabase(void)
+{
+	return CitusManagementDatabase;
+}
+
+
+void
+runCitusManagementQuery(char *query)
+{
+	if (managementCon == NULL)
+	{
+		int flags = 0;
+		managementCon = GetNodeUserDatabaseConnection(flags, LocalHostName,
+												  PostPortNumber,
+												  "ozan", "ozan");
+		RemoteTransactionBegin(managementCon);
+	}
+
+	SendRemoteCommand(managementCon, query);
+	ForgetResults(managementCon);
+}
+
+void
+cleanConnection(void)
```

Comment:
clean -> Clean, and name seems too generic. Also, should this close the connection?